app/src/main/java/com/example/tour_a_versity_v3/ui/MapsActivity.kt

```diff
@@ -35,39 +38,61 @@ class MapsActivity : AppCompatActivity(), OnMapReadyCallback, GoogleMap.OnInfoWi
         this.map = googleMap
         map.mapType = GoogleMap.MAP_TYPE_HYBRID
         map.setOnInfoWindowClickListener(this)
-        //set values for TUC
-        val tucLat = 39.13175
-        val tucLng = -84.51774
-        val zoomLevel = 17f
-        //create LatLng object for TUC
-        val tucLatLng = LatLng(tucLat, tucLng)
+
         //move camera to TUC & add marker
-        map.moveCamera(CameraUpdateFactory.newLatLngZoom(tucLatLng, zoomLevel))
-        tucMarker = map.addMarker(MarkerOptions()
-                .position(tucLatLng)
-                .title("TUC")
-                .snippet("Tangeman University Center")
-        )
+        moveCameraToTuc()
+        tucMarker = getTucMarker()
 
         //create another marker for CECH
-        val cechLat = 39.130315
-        val cechLng = -84.518680
-        val cechLatLng = LatLng(cechLat, cechLng)
-        cechMarker = map.addMarker(MarkerOptions()
-                .position(cechLatLng)
-                .title("CECH")
-                .snippet("College of Education, Criminal Justice, Human Services, and IT")
-        )
+        cechMarker = getCechMarker()
     }
+
     override fun onInfoWindowClick(p0: Marker?) {
         if (p0 == tucMarker){
-            val intent = Intent(this, TUC_Info::class.java)
+            val intent = Intent(this, InfoActivity::class.java)
             startActivity(intent)
         }
         else if (p0 == cechMarker){
-            val intent = Intent(this, CECH_Info::class.java)
+            val intent = Intent(this, InfoActivity::class.java)
             startActivity(intent)
         }
     }
 
+    /**
+     * Creates a marker object for TUC
+     */
+    private fun getTucMarker(): Marker {
+        val tucLat = 39.13175
+        val tucLng = -84.51774
+        val tucLatLng = LatLng(tucLat, tucLng)
+        return map.addMarker(MarkerOptions()
+            .position(tucLatLng)
+            .title("TUC")
+            .snippet("Tangeman University Center")
+        )
+    }
```

Comment:
These methods handle both data and ui logic, so technically they should be split between your activity and ViewModel. Again, I cannot recommend specific changes to your code since you don't have a DTO and a ViewModel class.